### ISSUE TITLE
Issue accessing column by index of ResultSet 

### ISSUE THREAD

**Issue by yevsh** [jdbc-json/jdbc-cb#57]
Hi,

I am not sure if this project is maintained, if yes, I think there is an issue with how the 'field' or 'column'

is retrieved by column index. e.g. getField(int columnNumber) in CBResultSet

field names are taken from signature json object from the result retrieved from the REST request to the sever.

The problem is. that fields of signature are not in the order of fields in query, the outcome you are getting wrong field for index.

For example, if query is
Select X,N,A from bucket;

signature will be:

{
“A”:“json”,
“N”:“json”,
“X”:“json”
}

**Comment by j-larson** [comment]
Yes, this is a known problem. I suggest you retrieve the fields by name.

**Comment by yevsh** [comment]
yes, but one of the requirement that I have is to retrieve by index :(

**Comment by yevsh** [comment]
In case I decide to implement this requirement,  the only solution would be to parse the N1QL query string (including alias names,functions, unions , joins and etc) and to match it to the signature object?

This would be extremely complex, but do you see another way to achieve that?

**Comment by j-larson** [comment]
No, that's what you'd have to do.
